src/main/java/org/jenkinsci/plugins/changeassemblyversion/ChangeAssemblyVersion.java

```diff
@@ -170,13 +241,14 @@ public boolean perform(AbstractBuild build, Launcher launcher, BuildListener lis
             StringWriter sw = new StringWriter();
             ex.printStackTrace(new PrintWriter(sw));
             listener.getLogger().println(sw.toString());
-            return false;
-        }
-        return true;
-    }
 
+            throw new AbortException(sw.toString());
+        }	    
+	    
+	}
+    
     @Extension
-    public static class Descriptor extends BuildStepDescriptor<Builder> {
+    public static class DescriptorImpl extends BuildStepDescriptor<Builder> {
```

Comment:
It formally breaks binary compatibility. Maybe it does not matter in the case of this plugin

Reply:
Without this rename, the plugin doesn't show in list in "Pipeline Syntax"

Reply:
@jglick @svanoort @abayer FYI, maybe some parsing logic specifics